posts/2023-12-18_admiral_1_0/admiral_1_0.qmd

```diff
@@ -0,0 +1,289 @@
+
+```{r}
+derive_vars_extreme_event(
+    adsl,
+    by_vars = exprs(STUDYID, USUBJID),
+    events = list(
+      event(
+        dataset_name = "adsl",
+        condition = !is.na(DTHDT),
+        set_values_to = exprs(LSTALVDT = DTHDT, DTHFL = "Y")
+      ),
+      event(
+        dataset_name = "adsl",
+        condition = !is.na(TRTEDT),
+        set_values_to = exprs(LSTALVDT = TRTEDT, DTHFL = "N")
+      )
+    ),
+    source_datasets = list(adsl = adsl),
+    order = exprs(LSTALVDT),
+    mode = "last",
+    new_vars = exprs(LSTALVDT = LSTALVDT, DTHFL = DTHFL)
+  )
+
+```
+
+Okay! We used a very small example to showcase how to find extreme observations and
+appending this information as new variables to our `ADSL` dataset.  Highly recommend checking
+out the more detailed example in the [function documentation](https://pharmaverse.github.io/admiral/reference/derive_vars_extreme_event.html) to see its true power!
+
+## `derive_var_merged_ef_msrc()`
+
+This function has some similarity to [derive_vars_extreme_event()](https://pharmaverse.github.io/admiral/reference/derive_vars_extreme_event.html), but now we are only looking at adding a single flag variable based on checking conditions across multiple datasets.
+
+We develop some _simple_ dummy data for `ADSL`, `CM` and `PR`. Our goal is flag patients who have `CMCAT = "ANIT-CANCER"` in the `CM` dataset or have records in the `PR` dataset. Any of participants who meet these condition will have our new variable `CANCTRFL` set as `Y`.
+
+```{r, message = FALSE, warning = FALSE}
+adsl <- tribble(
+    ~USUBJID, 
+    "1",      
+    "2",      
+    "3",      
+    "4",      
+  )
+
+cm <- tribble(
+    ~USUBJID, ~CMCAT,        ~CMSEQ,
+    "1",      "ANTI-CANCER",      1,
+    "1",      "GENERAL",          2,
+    "2",      "GENERAL",          1,
+    "3",      "ANTI-CANCER",      1
+  )
+
+ pr <- tribble(
+    ~USUBJID, ~PRSEQ,
+    "2",      1,
+    "3",      1
+  )
+```
+
+
+Now we have the argument `flag_events` that takes a list of objects where we define the conditions and datasets to check in.
+
+```{r}
+derive_var_merged_ef_msrc(
+    adsl,
+    flag_events = list(
+      flag_event(
+        dataset_name = "cm",
+        condition = CMCAT == "ANTI-CANCER"
+      ),
+      flag_event(
+        dataset_name = "pr"
+      )
+    ),
+    source_datasets = list(cm = cm, pr = pr),
+    by_vars = exprs(USUBJID),
+    new_var = CANCTRFL
+  )
+
+```
+Let's go! We searched over multiple datasets, `CM` and `PR` with multiple conditions and appended a new variable `CANCTRFL` to `ADSL` setting to "Y" if those conditions were met. 
+
+## `derive_vars_computed()`
+
+This function is very similar to [derive_vars_computed()](https://pharmaverse.github.io/admiral/reference/derive_vars_computed.html), but instead of adding observations we are going to add variable(s).  Very handy when wanting to add some additional variables to `ADSL`, e.g. baseline variables.
```

Comment:
```suggestion
This function is very similar to [derive_param_computed()](https://pharmaverse.github.io/admiral/reference/derive_param_computed.html), but instead of adding observations we are going to add variable(s).  Very handy when wanting to add some additional variables to `ADSL`, e.g. baseline variables.
```